internal/util/command.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+// The directory in which all of the exec.Cmds will be run.
+var commandsDir string
+
+// SetCommandsDir is the setter method for commandsDir.
+func SetCommandsDir(d string) {
+	commandsDir = d
+}
+
+// GcloudCommandBuild creates an exec.Cmd that calls the external gcloud executable. The exec.Cmd's name is set to
+// "gcloud" and the args are the ones provided in addition to a project flag and quiet flag.
+func GcloudCommandBuild(arg ...string) *exec.Cmd {
+	arg = append(arg, "--quiet")
+	cmd := exec.Command("gcloud", arg...)
+
+	return cmd
+}
+
+// ExecCommand executes an exec.Cmd. It redirects the command's stderr to this program's stderr and returns the output
+// in the form of a string. The command will be run in commandsDir.
+func ExecCommand(cmd *exec.Cmd) (out string, err error) {
+	cmd.Dir = commandsDir
+	cmd.Stderr = os.Stderr
```

Comment:
What about stdout?

Reply:
We're currently purposefully not connecting the command's stdout to a specific Writer so that we can extract it later through `cmd.Output`. Go currently doesn't allow for you tor redirect a command's output 
and later get it's output bytes later at the same time  (https://github.com/golang/go/blob/master/src/os/exec/exec.go#L536).

One idea I have to get around this is to do something like this:

```
var stdout bytes.Buffer
cmd.Stdout = io.MultiWriter(os.Stdout, &stdout)
```

and then use `cmd.Run()` instead of `cmd.Output()`. But this is only if we want to print out the command's stdout every time regardless of whether there's an error or not.

Reply:
There will be times, I think, where we don't want to print the stdout of a command, like when we call `gcloud auth print-identity-token`. So maybe there could be an argument specifying whether we should print the stodout of a command or not.